include/flatbuffers/flexbuffers.h

```diff
@@ -377,12 +379,44 @@ class Reference {
       case TYPE_NULL: return 0;
       case TYPE_STRING: return flatbuffers::StringToInt(AsString().c_str());
       case TYPE_VECTOR: return static_cast<int64_t>(AsVector().size());
+      case TYPE_BOOL: return static_cast<int64_t>(AsBool());
       default:
       // Convert other things to int.
       return 0;
     }
   }
 
+  bool AsBool() const {
+    if (type_ == TYPE_BOOL) {
```

Comment:
The body could be `if (type_ == TYPE_STRING) { .. } else return AsInt64() != 0`

Actually I'm not sure how I feel about this string special-case.. if the JSON contains "true" as a string, it be better to serialize it as a TYPE_BOOL, and then we wouldn't need this code here at all.